frontend/online-shop/src/components/ProductDetails.tsx

```diff
@@ -0,0 +1,87 @@
+import Paper from "@material-ui/core/Paper";
+import Grid from "@material-ui/core/Grid";
+import { useStyles } from "../styles/tableStyles";
+import AddShoppingCartIcon from "@material-ui/icons/AddShoppingCart";
+import { IconButton } from "@material-ui/core";
+import DeleteForeverIcon from "@material-ui/icons/DeleteForever";
+import { Link } from "react-router-dom";
+import ProductService from "../services/ProductService";
+
+const ProductDetails = (props: any) => {
+  const product = props.location.state.product;
```

Comment:
As this is a component that is directly tied to a route you shouldn't use props with it. Indeed when you come to this page from the products list page you will be able to send the product and retrieve it from the props. How about when you try to access directly http://localhost:4200/products/1? Also there is a method called useParams() that can retrieve the path parameters object without using props

Reply:
You should retrieve the product id from the path and make a service call to retrieve the product with the respective id.